rust/src/chain_crypto/bech32.rs

```diff
@@ -0,0 +1,79 @@
+use bech32::{Error as Bech32Error, FromBase32, ToBase32};
+use std::error::Error as StdError;
+use std::fmt;
+use std::result::Result as StdResult;
+
+pub type Result<T> = StdResult<T, Error>;
+
+pub trait Bech32 {
+    const BECH32_HRP: &'static str;
+
+    fn try_from_bech32_str(bech32_str: &str) -> Result<Self>
+    where
+        Self: Sized;
+
+    fn to_bech32_str(&self) -> String;
+}
+
+pub fn to_bech32_from_bytes<B: Bech32>(bytes: &[u8]) -> String {
+    bech32::encode(B::BECH32_HRP, bytes.to_base32())
+        .unwrap_or_else(|e| panic!("Failed to build bech32: {}", e))
+        .to_string()
+}
+
+pub fn try_from_bech32_to_bytes<B: Bech32>(bech32_str: &str) -> Result<Vec<u8>> {
+    let (hrp, bech32_data) = bech32::decode(bech32_str)?;
```

Comment:
Unfortunate we had to downgrade this for this PR. Maybe create an issue to upgrade the bech32 library and reference this upstream PR so that we know how to fix this up quickly in the future? https://github.com/input-output-hk/chain-libs/commit/53e791cdb7ba138dfaf6b89e67c099e73a229baa#diff-9bc73802da0f8d3c0fbfaae52e2e5701